### PR TITLE
Changing rendering

### PR DESCRIPTION
**Pull request by ToBinio**
changing rendering so that the render_pass can be reused

### CODE REVIEW

File: src/shapes.rs [hunk 1]
```diff
@@ -175,7 +122,73 @@ impl Shape for Quadrat {
...
+        indices_buffer
+    }
+}
+
+impl Shape for Quadrat {
+    fn get_vertex_buffer(&self) -> &Buffer {
+        &self.vertex_buffer
+    }
+
+    fn update_vertex_buffer(&mut self, device: &Device) {
```
Comment (JonasPfeifer05) [inline src/shapes.rs:174]: So on each draw call the buffer should be created completely new?

**Review by JonasPfeifer05 — CHANGES_REQUESTED** [review]

**Comment by JonasPfeifer05** [comment]
Why is that so?

**Comment by ToBinio** [comment]
after now removing the update handleling in the draw function i wondert if it is nessary to have an update function delcated in the trait. over all it is currently not need in the trait but it probaly is a good reminder to have it in there so you dont forget that you need to update the buffers... what are your thoughts?

**Comment by JonasPfeifer05** [comment]
Yea I think we could split it up in something like a StaticComponent/shape and DynamicComponent/shape if we later need to update the buffers. But now we dont need it

**Pull request merged**